### PR TITLE
request to merge vocal-contour

### PR DESCRIPTION
**Pull request by yjlolo**
This is to add `vocal-contour` which adapts from the [original repo](https://github.com/s603122001/Vocal-Melody-Extraction).

- The PR provides the essential functionality to `transcribe`, `generate_feature`, and `train` for frame-level vocal melody extraction, which also serves as the foundation for the note-level vocal melody transcription under development.
- One missing feature is preprocessing of `MedleyDB`; so training the model with data mixing `MIK-1K` and `MedleyDB` as in the [paper](https://archives.ismir.net/ismir2018/paper/000286.pdf) is disabled currently. Despite the missing feature, the model checkpoint `Seg` from the [link to checkpoints provided in the original repo](https://drive.google.com/file/d/13kApyZ5lJEGE5CDwaeEuxVuw9sZy_xae/view) is implemented and provided in this PR.
- Another missing feature is the accommodation of different variations of models (e.g. progressive ones, refer to the [paper](https://archives.ismir.net/ismir2018/paper/000286.pdf) for more detail). The implemented model is `Seg`, which performs the best according to our experiments.
- Documentation is to be updated.

### CODE REVIEW

**Review by BreezeWhite — CHANGES_REQUESTED** [review]
Some more things to be added:

- [ ] Add unit tests. At least add test of loading checkpoints like [here](https://github.com/Music-and-Culture-Technology-Lab/omnizart/blob/master/tests/music/test_app.py).
- [ ] Make sure passing the `general-check` before merging. Seems like there are still problems with the dependencies.

Great thanks to your huge contributions^^.

File: omnizart/vocal_contour/app.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        f0 = r[:, 1].astype(float)
+        t = np.arange(len(f0)) * model_settings.feature.hop_size
+        y = sonify.pitch_contour(t, f0, model_settings.feature.sampling_rate)
+
+        if output is not None:
+            base = os.path.basename(input_audio)
+            filename, ext = os.path.splitext(base)
+            f0_out = filename + '_f0' + '.txt'
+            audio_trans = filename + '_trans' + '.wav'
+            save_to = output
```
[the remark this answers is not in the document]
Comment (yjlolo) [reply omnizart/vocal_contour/app.py:87]: This was to be consistent with [`music.app`](https://github.com/Music-and-Culture-Technology-Lab/omnizart/blob/d27412bdc57da9f646a2ae244796bb5569af13ff/omnizart/music/app.py#L113).

File: omnizart/cli/vocal_contour/train_model.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from omnizart.vocal_contour import app
+from omnizart.setting_loaders import VocalFrameSettings
+
+
+click.option = partial(click.option, show_default=True)
+
+
+@click.command()
+@add_common_options(COMMON_TRAIN_MODEL_OPTIONS)
+@click.option("-t", "--timesteps", help="Time width of each input feature", type=int, default=128)
```
[the remark this answers is not in the document]
Comment (yjlolo) [reply omnizart/cli/vocal_contour/train_model.py:15]: I was following the same line from [`music`](https://github.com/Music-and-Culture-Technology-Lab/omnizart/blob/d27412bdc57da9f646a2ae244796bb5569af13ff/omnizart/cli/music/train_model.py#L46). 
I suppose there is a better way to pass the default values from the setting, and at the same time allow the optional argument as the input?

File: omnizart/cli/vocal_contour/generate_feature.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    show_default=True
+)
+@click.option(
+    "-s",
+    "--sampling-rate",
+    help="Adjust input sampling rate to this value.",
+    type=int,
+    default=16000,
+    show_default=True
+)
```
[the remark this answers is not in the document]
Comment (yjlolo) [reply omnizart/cli/vocal_contour/generate_feature.py:28]: The purpose here is to show which variables could be critical for resulted transcription accuracy.
(Re-)setting default values here is to be able to `show_default` for the users when `--help` is called, so they can assign to other values if needed. What do you think?

**Review by BreezeWhite — APPROVED** [review]

**Pull request merged**